### PR TITLE
SPM-1061: Manager returns 500 (unknonw system)

### PR DESCRIPTION
**Pull request by AlesKas**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by semtexzv — APPROVED** [review]
Nice and simple fix

File: base/database/utils.go [hunk 1]
```diff
@@ -96,3 +97,8 @@ func PluckInt(tx *gorm.DB, columnName string) int {
 	}
 	return val
 }
+
+func IsValidUUID(uuid string) bool {
```
Comment (Josca) [inline base/database/utils.go:101]: Great, but please put this function to "base/utils/core.go", so we can call it as `utils.IsValidUUID(uuid string) bool`.

File: manager/controllers/system_advisories_export_test.go [hunk 2]
```diff
@@ -41,3 +41,14 @@ func TestSystemAdvisoriesExportCSV(t *testing.T) {
 	assert.Equal(t, "id,description,public_date,synopsis,advisory_type,severity,cve_count", lines[0])
 	assert.Equal(t, "RH-1,adv-1-des,2016-09-22T16:00:00Z,adv-1-syn,1,,0", lines[1])
 }
+
+func TestUnknownSystemAdvisoriesExport(t *testing.T) {
```
Comment (Josca) [inline manager/controllers/system_advisories_export_test.go:45]: Please keep test methods name convention "specific part to the end" - here TestSystemAdvisoriesExportUnknown. It makes it easiers to find tests for the same endpoint when printed in logs. Also for other tests.

File: manager/controllers/system_advisories.go [hunk 3]
```diff
@@ -77,6 +77,11 @@ func SystemAdvisoriesHandler(c *gin.Context) {
 		return
 	}
 
+	if !(database.IsValidUUID(inventoryID)) {
```
Comment (Josca) [inline manager/controllers/system_advisories.go:80]: You can omit these external brackets - use just `if !utils.IsValidUUID(inventoryID) {`.

File: base/database/utils.go [hunk 4]
```diff
@@ -96,3 +97,8 @@ func PluckInt(tx *gorm.DB, columnName string) int {
...
+	r := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$")
```
Comment (Josca) [inline base/database/utils.go:102]: This regex can be created in "var" no to be created during each function call. Let's call it e.g. uuidRegex.

~~~golang
var uuidRegex = regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$")

func IsValidUUID(uuid string) bool {
	return uuidRegex.MatchString(uuid)
}

~~~

**Review by Josca — CHANGES_REQUESTED** [review]
Great, good job. Just some small updates required.

**Review by Josca — APPROVED** [review]
Looks good. Thanks :+1:

**Pull request merged**